internal/Yesod/Session/Persist/Freeze/Action.hs

```diff
@@ -0,0 +1,29 @@
+module Yesod.Session.Persist.Freeze.Action
+  ( disableSessionManagement
+  , assignSessionFreeze
+  ) where
+
+import Yesod.Session.Persist.Prelude
+
+import Yesod.Core (MonadHandler (liftHandler), getYesod)
+import Yesod.Session.Persist.Freeze.Type
+import Yesod.Session.Persist.Options
+
+-- | Indicate whether the session should be frozen for the handling
+--   of the current request
+--
+-- At the end of the request handler, if the value is 'Just', no
+-- database actions will be performed and no cookies will be set.
+assignSessionFreeze
+  :: (MonadHandler m, HasSessionEmbeddings (HandlerSite m))
```

Comment:
So, if the idea is `Yesod.Session.{Something}` as a `Something`-backed session thing for Yesod, then wouldn't it make more sense for functions like this, which don't depend on the `Something`, to live in (e.g.) `Yesod.Session.Freeze.Action`?

It seems like you've constructing thing such that a lot of functionality is un-related to the `Persist` part of it, which seems good for understanding and test-ability or whatever. Should probably convey / take advantage of that in the module hierarchy?

Reply:
Although, I guess I'm not sure what `HasSessionEmbeddings` means. If it's hiding something persistent-related, then I guess my comment doesn't apply.

Reply:
Cool, did a reorganization to better distinguish which things pertain specifically to peristent, yesod, sessions in general, or none of the above.